### ISSUE TITLE
actions-secret-parser: Invalid 'main' field in package.json

### ISSUE THREAD

**Issue by tommyhtran** [microsoft/pipelines-appservice-lib#71]
When running the **azure/webapps-deploy@v2** action, I received the following error:
> (node:2000) [DEP0128] DeprecationWarning: Invalid 'main' field in '/home/runner/work/_actions/azure/webapps-deploy/v2/node_modules/actions-secret-parser/package.json' of 'lib/index.js'. Please either fix that or report it to the module author

The configuration seems correct in this repository because **/packages/secret-parser/tsconfig.json** outputs into a **/lib** directory and the 'main' field in **package.json** is referencing that directory.

The problem is that the npm package **actions-secret-parser** is different from the configuration. **index.js** is in the root directory, the **/lib** directory does not exist. See https://www.npmjs.com/package/actions-secret-parser?activeTab=explore

**Comment by ctrlplusb** [comment]
I've just started to hit the same issue.

**Comment by molily** [comment]
Hi, does anyone know a way to work around this problem?

**Comment by StefanLobbenmeierObjego** [comment]
yeah the fix you suggested is in https://github.com/microsoft/pipelines-appservice-lib/commit/4981bed9e74c5c3dc607482314e953009c07aec4
